docs/DeveloperGuide.md

```diff
@@ -4,10 +4,168 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design & Implementation
 
+### Architecture
+The Calendar application is designed with a modular architecture, consisting of the following components:
+
+- **UI**: The `UiRenderer` class handles the rendering of the calendar views (week and month) to the console.
+- **Logic**: The `Main` class acts as the central logic component, handling user input and dispatching commands to the appropriate components.
+- **Data**: The `TaskManager` class manages the tasks and their corresponding dates, providing methods for adding, updating, and deleting tasks.
+- **Storage**: The `Storage` class handles the persistence of tasks by reading from and writing to a file.
+- **Time**: The `View` class and its subclasses (`WeekView` and `MonthView`) manage the rendering and navigation of the week and month views.
+
+The following sequence diagram illustrates the high-level interaction between the components when executing a command:
+
+```puml
+@startuml
+participant Main
+participant TaskManager
+participant WeekView
+participant MonthView
+participant Storage
+
+Main -> TaskManager : executeCommand()
+activate TaskManager
+TaskManager -> WeekView : getWeekView()
+TaskManager -> MonthView : getMonthView()
+TaskManager -> Storage : loadTasksFromFile()
+TaskManager <-- Storage : tasksFromFile
+TaskManager -> WeekView : printView()
+TaskManager -> MonthView : printView()
+deactivate TaskManager
+@enduml
+```
```

Comment:
Not sure if this will generate required UML diagrams. It might be better if we draw with an application / by hand and upload the image instead.

Reply:
Agreed, please generate the diagram first then insert the image into this file.